### PR TITLE
Moved Github integration into a class

### PR DESCRIPTION
**Pull request by DebitCardz**
Added it to SynthesisClient

### CODE REVIEW

**Comment by DebitCardz** [comment]
It may be possible for integration#resync to return a JSON object with data such as

- Issues channel
- Total comments
- Total issues/prs
For now it'll just return the new issues channel so we can save that somewhere(SynthesisClient?).

File: src/client.ts [hunk 1]
```diff
@@ -1,22 +1,27 @@
 import { CommandClient, CommandClientOptions, event } from "../deps.ts";
 import SyncCommand from "./commands/SyncCommand.ts";
 import Config from "./types/Config.ts";
+import GithubIntegration from "./github/integration.ts";
 
 export default class SynthesisClient extends CommandClient {
   // cache here
   public config: Config;
 
+  public github: GithubIntegration;
```
Comment (CoreyShupe) [inline src/client.ts:10]: make this readonly private along with a getter, and remove the line between config and github

File: src/commands/SyncCommand.ts [hunk 2]
```diff
@@ -1,111 +1,19 @@
...
+import { Command, CommandContext, Embed, User } from "../../deps.ts";
+import SynthesisClient from "../client.ts";
 
 export default class SyncCommand extends Command {
   name = "sync";
 
   async execute(ctx: CommandContext) {
-    const start = Date.now();
-    const config = requireSynthesisConfig(ctx);
+    const client = ctx.client as SynthesisClient;
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:8]: this should be handled in functions for proper assertion of the SynthesisClient

File: src/commands/SyncCommand.ts [hunk 3]
```diff
@@ -1,111 +1,19 @@
...
-        await guild.channels.delete(channel.id);
-      }
-    }
-
-    return await guild.createChannel({
-      name: config.discord.channels.issues.name,
-      parent: config.discord.channels.issues.parent,
-      type: ChannelTypes.GUILD_TEXT,
-    });
+    await client.github.resync(config, ctx.guild);
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:15]: github integration should solely be a linker between TS and Github, it should not supply logic

File: src/github/integration.ts [hunk 4]
```diff
@@ -1,35 +1,119 @@
-import { Octokit } from "../../deps.ts";
-import { config } from "../types/Config.ts";
+import { ChannelTypes, Guild, GuildChannel, Octokit } from "../../deps.ts";
+import Config, { config } from "../types/Config.ts";
```
Comment (CoreyShupe) [inline src/github/integration.ts:2]: remove the `config` decleration here

File: src/github/integration.ts [hunk 5]
```diff
@@ -1,35 +1,119 @@
...
-}
+  constructor(config: Config) {
+    this.octokit = new Octokit({ auth: config.github.token });
+  }
+
+  /**
+   * Resync the discord with all github issues and pull requests.
+   * @returns Issues channel.
+   */
+  async resync(config: Config, guild: Guild): Promise<GuildChannel> {
```
Comment (CoreyShupe) [inline src/github/integration.ts:23]: again, this class should not be handling logic of issues, it should merely be an interface to github

File: src/github/integration.ts [hunk 6]
```diff
@@ -1,35 +1,119 @@
...
-    {
+    return issuesChannel;
+  }
+
+  // We can possibly move this somewhere else if wanted.
+  /**
+   * Remove the existing issues channel and replace it with a new one.
+   * @returns New issues channel.
+   */
+  private async generateIssuesChannel(
```
Comment (CoreyShupe) [inline src/github/integration.ts:74]: this shouldn't be handled by github integration

File: src/util/functions.ts [hunk 7]
```diff
@@ -15,4 +15,4 @@ export const requireSynthesisConfigFromClient = (
 };
 
 export const requireSynthesisConfig = (ctx: CommandContext): Config =>
```
Comment (CoreyShupe) [inline src/util/functions.ts:17]: we should also have something which returns the SynthesisClient or properly throws in this file

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: src/client.ts (hunk 1 above)
Comment (DebitCardz) [reply src/client.ts:10]: We can make this a public readonly instead.

File: src/client.ts [hunk 8]
```diff
@@ -1,22 +1,27 @@
...
     this.config = JSON.parse(
       Deno.readTextFileSync("../secrets/config.json"),
     );
+
+    this.github = new GithubIntegration(this.config);
   }
 
   @event()
-  ready(): void {
+  async ready() {
```
Comment (CoreyShupe) [inline src/client.ts:24]: why did we remove the explicit `:void` and why does this need to be async

File: src/commands/SyncCommand.ts [hunk 9]
```diff
@@ -32,69 +32,95 @@ export default class SyncCommand extends Command {
       `Synchronizing issues with **${config.github.repo}**.`,
     );
 
-    // TODO: We can store this somewhere else later, for now this works.
-    const issueChannel = await this.resetIssuesChannel(config, ctx.guild);
+    const information = await this.resync(client.github, config, ctx.guild);
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:35]: why don't we just pass client into this and get `client.github` and `client.config`

File: src/commands/SyncCommand.ts [hunk 10]
```diff
@@ -32,69 +32,95 @@ export default class SyncCommand extends Command {
...
+   * Resync the discord with all github issues and pull requests.
+   * @returns Information about the sync request.
+   */
+  private async resync(
+    github: GithubIntegration,
+    config: Config,
+    guild: Guild,
+  ): Promise<SyncResult> {
+    const issuesChannel = await this.generateIssuesChannel(config, guild);
+    const client = issuesChannel.client;
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:57]: this seems like a messy way to get the client, just pass the client in

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: src/client.ts (hunk 8 above)
Comment (DebitCardz) [reply src/client.ts:24]: This was an old change I made when testing something that I forgot to revert.

File: src/commands/SyncCommand.ts (hunk 9 above)
Comment (DebitCardz) [reply src/commands/SyncCommand.ts:35]: I felt like it'd just be cleaner to only pass exactly whatever we needed, we don't need the entire client scope so I only passed the things we needed.
Let me know if you'd like this changed.

Comment (DebitCardz) [reply src/commands/SyncCommand.ts:35]: Actually this has been changed in the latest commit to just pass client.

File: src/commands/SyncCommand.ts (hunk 10 above)
Comment (DebitCardz) [reply src/commands/SyncCommand.ts:57]: Yeah agreed, fixed this.

**Comment by CoreyShupe** [comment]
LGTM

**Comment by DebitCardz** [comment]
WGTM

**Pull request merged**